internal/testcli/golden.go

```diff
@@ -0,0 +1,230 @@
+package testcli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/databricks/cli/internal/testutil"
+	"github.com/databricks/cli/libs/iamutil"
+	"github.com/databricks/cli/libs/testdiff"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/stretchr/testify/require"
+)
+
+var OverwriteMode = os.Getenv("TESTS_OUTPUT") == "OVERWRITE"
+
+func ReadFile(t testutil.TestingT, ctx context.Context, filename string) string {
+	data, err := os.ReadFile(filename)
+	if os.IsNotExist(err) {
+		return ""
+	}
+	require.NoError(t, err)
+	// On CI, on Windows \n in the file somehow end up as \r\n
+	return NormalizeNewlines(string(data))
+}
+
+func captureOutput(t testutil.TestingT, ctx context.Context, args []string) string {
+	t.Logf("run args: [%s]", strings.Join(args, ", "))
+	r := NewRunner(t, ctx, args...)
+	stdout, stderr, err := r.Run()
+	require.NoError(t, err)
+	out := stderr.String() + stdout.String()
+	return ReplaceOutput(t, ctx, out)
+}
+
+func WriteFile(t testutil.TestingT, filename, data string) {
+	t.Logf("Overwriting %s", filename)
+	err := os.WriteFile(filename, []byte(data), 0o644)
+	require.NoError(t, err)
+}
+
+func RequireOutput(t testutil.TestingT, ctx context.Context, args []string, expectedFilename string) {
+	_, filename, _, _ := runtime.Caller(1)
```

Comment:
Could you add a comment or separate it into a named function to describe what you're trying to achieve? From what I understood you're trying to get the location of the file where the test is located, correct?

Reply:
Good callout. I removed this part. Instead I added a helper to calculate path to testdata in case or changed directory.